pkgs/applications/networking/remote/aws-workspaces/default.nix

```diff
@@ -45,6 +45,10 @@ stdenv.mkDerivation rec {
     ${dpkg}/bin/dpkg -x $src $out
   '';
 
+  preFixup = ''
+    patchelf --replace-needed liblttng-ust.so.0 liblttng-ust.so $out/bin/libcoreclrtraceptprovider.so
```

Comment:
Why is the shared object under bin/?

Would it be a better idea to add a symlink for liblttng-ust.so to prevent further changes like this?

Reply:
[No](https://github.com/NixOS/nixpkgs/pull/136544#issuecomment-916709726).

If programs don't care about the version of liblttng-ust at all, they should use the unversioned library.  If they do care, but are compatible with the new version regardless, they should be updated to look for that version.  If an update isn't forthcoming, it should be the packages that need the hacks that have the hacks applied to them — if we were to put a version hack in the lttng-ust package, that would risk breaking entirely correct applications that know they're not compatible with the old version and specifically ask for the new one.